### PR TITLE
Feature/add unity event step

### PR DESCRIPTION
**Pull request by VladmirCSouza**

### CODE REVIEW

File: Scripts/Editor/Core/Steps/CallMethodAnimationStepCustomEditor.cs [hunk 1]
```diff
@@ -0,0 +1,31 @@
+﻿using BrunoMikoski.AnimationSequencer;
+using UnityEditor;
+using UnityEngine;
+
```
Comment (brunomikoski) [inline Scripts/Editor/Core/Steps/CallMethodAnimationStepCustomEditor.cs:3]: missing namespace.

File: Scripts/Runtime/Core/Steps/CallMethodAnimationStep.cs [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            
+            return $"{index}. {DisplayName}: {persistentTargetNames}";
+        }
+
+        public override void Complete()
+        {
+            methodToCall?.Invoke();
+        }
+    }
+}
```
Comment (brunomikoski) [inline Scripts/Runtime/Core/Steps/CallMethodAnimationStep.cs]: missing new line at the end of file.

**Review by brunomikoski — CHANGES_REQUESTED** [review]
Also check the Codacy report here:
https://app.codacy.com/gh/brunomikoski/Animation-Sequencer/pullRequest?prid=7279848

File: Scripts/Editor/Core/Steps/CallMethodAnimationStepCustomEditor.cs (hunk 1 above)
Comment (VladmirCSouza) [reply Scripts/Editor/Core/Steps/CallMethodAnimationStepCustomEditor.cs:3]: Here: 4b638bd3c6d2d7440fc617cc20949c07c107d01e

File: Scripts/Runtime/Core/Steps/CallMethodAnimationStep.cs (hunk 2 above)
Comment (VladmirCSouza) [reply Scripts/Runtime/Core/Steps/CallMethodAnimationStep.cs]: Here: 4b638bd3c6d2d7440fc617cc20949c07c107d01e

**Comment by VladmirCSouza** [comment]
Also applied the changes reported on Codacy. :)
4b638bd3c6d2d7440fc617cc20949c07c107d01e

File: Scripts/Runtime/Core/Steps/CallMethodAnimationStep.cs [hunk 3]
```diff
@@ -10,8 +10,11 @@ public class CallMethodAnimationStep : AnimationStepBase
         private UnityEvent methodToCall;
         
         public override string DisplayName => "Method Call Step";
-        
-        public override float Duration { get; }
+
+        public override float Duration
+        {
+            return 0;
```
Comment (brunomikoski) [inline Scripts/Runtime/Core/Steps/CallMethodAnimationStep.cs]: It's missing the get, right? 
It should be 
`public override string Duration => 0;`

File: Scripts/Runtime/Core/Steps/CallMethodAnimationStep.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
+using System;
+using UnityEngine;
+using UnityEngine.Events;
+
+namespace BrunoMikoski.AnimationSequencer
+{
+    public class CallMethodAnimationStep : AnimationStepBase
```
Comment (brunomikoski) [inline Scripts/Runtime/Core/Steps/CallMethodAnimationStep.cs]: also its should be a sealed class for now, so `public sealed class CallMethodAnimationStep : AnimationStepBase`

File: Scripts/Runtime/Core/Steps/CallMethodAnimationStep.cs (hunk 3 above)
Comment (VladmirCSouza) [reply Scripts/Runtime/Core/Steps/CallMethodAnimationStep.cs]: Yes! Forget to push this in the previous commit. 
Should be OK now. :D

**Review by brunomikoski — APPROVED** [review]

**Pull request merged**